test/gtest/common/test.h

```diff
@@ -169,72 +171,94 @@ class entities_storage {
 /*
  * Helper macro
  */
-#define UCS_TEST_(test_case_name, test_name, parent_class, parent_id, num_threads, ...)\
-class GTEST_TEST_CLASS_NAME_(test_case_name, test_name) : public parent_class {\
- public:\
-  GTEST_TEST_CLASS_NAME_(test_case_name, test_name)() {\
+#define UCS_TEST_(test_case_name, test_name, parent_class, parent_id, num_threads, skip_cond, ...) \
+class GTEST_TEST_CLASS_NAME_(test_case_name, test_name) : public parent_class { \
+ public: \
+  GTEST_TEST_CLASS_NAME_(test_case_name, test_name)() { \
      set_num_threads(num_threads); \
      UCS_PP_FOREACH(UCS_TEST_SET_CONFIG, _, __VA_ARGS__) \
   } \
- private:\
-  virtual void test_body();\
-  static ::testing::TestInfo* const test_info_;\
-  GTEST_DISALLOW_COPY_AND_ASSIGN_(\
-      GTEST_TEST_CLASS_NAME_(test_case_name, test_name));\
-};\
-\
-::testing::TestInfo* const GTEST_TEST_CLASS_NAME_(test_case_name, test_name)\
-  ::test_info_ =\
-    ::testing::internal::MakeAndRegisterTestInfo(\
-        #test_case_name, \
-        (num_threads == 1) ? #test_name : #test_name "/mt_" #num_threads, \
-        "", "", \
-        (parent_id), \
-        parent_class::SetUpTestCase, \
-        parent_class::TearDownTestCase, \
-        new ::testing::internal::TestFactoryImpl<\
-            GTEST_TEST_CLASS_NAME_(test_case_name, test_name)>);\
-void GTEST_TEST_CLASS_NAME_(test_case_name, test_name)::test_body()
+ private: \
+  virtual void check_skip_test() { \
+     if (skip_cond) { \
+         UCS_TEST_SKIP_R(UCS_PP_MAKE_STRING(skip_cond)); \
+     } \
+  } \
+  virtual void test_body(); \
+  static ::testing::TestInfo* const test_info_; \
+  GTEST_DISALLOW_COPY_AND_ASSIGN_( \
+      GTEST_TEST_CLASS_NAME_(test_case_name, test_name)); \
+    static int AddToRegistry() { \
```

Comment:
why do we need to define AddToRegistry()? seems we just need to add check_skip_test() ?

Reply:
sorry, this was from the previous implementation where we check `skip_cond` argument inside `AddToRegistru()`

`AddToRegistru()` removed